unicode_data_gen/src/lib.rs

```diff
@@ -0,0 +1,201 @@
+
+/// Exports ICU data provider as Rust code into the `out` directory.
+pub fn generate(out: std::path::PathBuf) {
+    let icu4x_source_provider = SourceDataProvider::new();
+
+    // Generate ICU4X data
+    {
+        // The directory for ICU baked data is `out/icu4x_data`.
+        let icu4x_data_dir = out.clone().join("icu4x_data");
+        std::fs::create_dir_all(&icu4x_data_dir).unwrap();
+
+        ExportDriver::new(
+            [DataLocaleFamily::single(locale!("en").into())],
+            DeduplicationStrategy::Maximal.into(),
+            LocaleFallbacker::new_without_data(),
+        )
+        .with_markers([
+            icu_properties::provider::PropertyNameShortScriptV1::INFO,
+            icu_segmenter::provider::SegmenterBreakGraphemeClusterV1::INFO,
+            icu_segmenter::provider::SegmenterBreakWordOverrideV1::INFO,
+            icu_segmenter::provider::SegmenterLstmAutoV1::INFO,
+            icu_segmenter::provider::SegmenterBreakWordV1::INFO,
+            icu_segmenter::provider::SegmenterBreakLineV1::INFO,
+            icu_normalizer::provider::NormalizerNfcV1::INFO,
+            icu_normalizer::provider::NormalizerNfdDataV1::INFO,
+            icu_normalizer::provider::NormalizerNfdTablesV1::INFO,
+        ])
+        .export(
+            &icu4x_source_provider.clone(),
+            icu_provider_export::baked_exporter::BakedExporter::new(icu4x_data_dir.clone(), {
+                let mut o = icu_provider_export::baked_exporter::Options::default();
+                o.overwrite = true;
+                o.use_separate_crates = true;
+                o.pretty = true;
+                o
+            })
+            .unwrap(),
+        )
+        .expect("Datagen should be successful");
+        std::fs::write(
+            icu4x_data_dir.clone().join("mod.rs"),
+            COPYRIGHT_HEADER.to_string()
+                + "\n#![allow(clippy::allow_attributes_without_reason)]\n"
+                + &std::fs::read_to_string(icu4x_data_dir.clone().join("mod.rs")).unwrap(),
+        )
+        .unwrap();
+    }
+
+    // Generate `CompositePropsV1` data
+    {
+        let custom_source_provider = CompositePropsProvider::new(icu4x_source_provider.clone());
+
+        // The directory for the composite data is `out/composite`.
+        let custom_data_dir = out.clone().join("composite");
+        std::fs::create_dir_all(&custom_data_dir).unwrap();
+        ExportDriver::new(
+            [DataLocaleFamily::single(locale!("en").into())],
+            DeduplicationStrategy::None.into(),
+            LocaleFallbacker::new_without_data(),
+        )
+        .with_markers([CompositePropsV1::INFO])
+        .export(
+            &ForkByMarkerProvider::new(icu4x_source_provider.clone(), custom_source_provider),
+            BlobExporter::new_with_sink(Box::new(
+                std::fs::File::create(custom_data_dir.clone().join("composite.postcard")).unwrap(),
```

Comment:
Question: Is there a reason you're exporting your custom `CompositePropsV1` as postcard instead of baked, since most of your other data is exported as baked? Have you tried plugging it into the baked exporter?